cmd/kubectl-gadget/utils/trace.go

```diff
@@ -287,36 +434,228 @@ RetryLoop:
 			}
 		}
 
+		printErrorFunction := func(format string, args ...interface{}) { fmt.Fprintf(os.Stderr, "[ERR]"+format+"\n", args...) }
+
+		// Print errors even if some nodes succeeded.
+		defer printTraceFeedback(printErrorFunction, nodeErrors)
+
 		// Don't print warnings if at least one node succeeded. This avoids showing
 		// warnings together with the actual output generated by other nodes.
 		if successNodeCount == 0 {
-			printTraceFeedback(contextLogger.Warningf, nodeWarnings)
+			printWarningFunction := func(format string, args ...interface{}) { fmt.Fprintf(os.Stderr, "[WARN] "+format+"\n", args...) }
```

Comment:
I am not sure I like that prefix `[WARN/ERR]` because not all the logs have it:
```
$ ./kubectl-gadget process-collector -n non-exixting-namespace
[WARN] Failed to run the gadget on all nodes: No container matches the requested filter
Failed to run the gadget on all nodes: None of them succeeded
```
Isn't there a way to make what we were doing with the `contextLogger` but at `main` level so that, as you said, we avoid passing the `contextLogger` to every function? I am thinking of something like what we are doing with the `params utils.CommonFlags` which are accessible from everywhere ([link](https://github.com/kinvolk/inspektor-gadget/blob/4dc3ae6c642e195117a84204c863bef41848bc31/cmd/kubectl-gadget/main.go#L27)).

@mauriciovasquezbernal any suggestion?

Reply:
I think `printTraceFeedback` is too complex now. Why not just to print the messages without the prefix? (we can remove the callback parameter and so on...)
If we really want to print the prefix I think we should use a logger everywhere, otherwise we'll reimplementing the wheel.